File: app/models/post.rb
class Post < ActiveRecord::Base
  attr_accessible :author_id, :recipient_id, :text

  validates :author_id, :recipient_id, :text, presence: true

  belongs_to(
    :author,
    class_name: "User",
    foreign_key: :author_id,
    primary_key: :id
  )

  belongs_to(
    :recipient,
    class_name: "User",
    foreign_key: :recipient_id,
    primary_key: :id
  )

  has_many(
    :comments,
    as: :commentable,
    foreign_key: :commentable_id,
    primary_key: :id,
    dependent: :destroy
  )

  has_many(
    :likes,
    as: :likeable,
    foreign_key: :likeable_id,
    primary_key: :id,
    dependent: :destroy
  )

  # note: is this necessary? or right?
  has_many(
    :liking_users,
    through: :likes,
    source: :user
  )

  def liked_by?(user)
    self.likes.where(user_id: user).any?
  end
end

File: app/assets/javascripts/models/post.js
Facebook.Models.Post = Backbone.Model.extend({
  parse: function (resp) {
    resp.author = new Facebook.Models.User(resp["author"]);
    resp.recipient = new Facebook.Models.User(resp["recipient"]);
    resp.comments = new Facebook.Collections.Comments(resp["comments"], {
      commentable: this
    });
  	resp.likes = new Facebook.Collections.Likes(resp["likes"], {
  		likeable: this
  	});

    return resp;
  },

  urlRoot: "/posts",

  toJSON: function () {
    var data = { "user_id" : this.get('recipient').get('id'),
                 "post" : { "text" : this.get('text') }};

    return data;
  }
});
